### PR TITLE
feat(sol): sol-component

### PR DESCRIPTION
**Pull request by Mahendra-intel**

### CODE REVIEW

File: src/app/devices/sol/sol.component.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+export class SolComponent implements OnInit {
+  @Input() solStatus: any = false
+  @Output() deviceStatus: EventEmitter<number> = new EventEmitter<number>()
+  public term: Terminal | any
+  public container!: HTMLElement | any
+  public terminal: AmtTerminal | any
+  public logger: ConsoleLogger = new ConsoleLogger(LogLevel.ERROR)
+  public redirector: AMTRedirector | any
+  public server: string = environment.mpsServer.replace('https://', '')
+  public uuid: string | any = this.route.snapshot.paramMap.get('id')
```
Comment (rsdmike) [inline src/app/devices/sol/sol.component.ts:22]: this should be retrieved from activatedRoute in ngOnInit()

File: src/app/devices/sol/sol.component.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+@Component({
+  selector: 'app-sol',
+  templateUrl: './sol.component.html',
+  styleUrls: ['./sol.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class SolComponent implements OnInit {
+  @Input() solStatus: any = false
+  @Output() deviceStatus: EventEmitter<number> = new EventEmitter<number>()
+  public term: Terminal | any
```
Comment (rsdmike) [inline src/app/devices/sol/sol.component.ts:16]: why do all of these have "| any"?

File: src/app/devices/sol/sol.component.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+import { environment } from 'src/environments/environment'
+
+@Component({
+  selector: 'app-sol',
+  templateUrl: './sol.component.html',
+  styleUrls: ['./sol.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class SolComponent implements OnInit {
+  @Input() solStatus: any = false
```
Comment (rsdmike) [inline src/app/devices/sol/sol.component.ts:14]: this should be of type bool

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: src/app/devices/sol/sol.component.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    this.dataProcessor = null
+    this.term = null
+  }
+
+  handleKeyPress = (domEvent: any): void => {
+    this.terminal.TermSendKeys(domEvent)
+  }
+
+  ngOnDestroy (): void {
+    this.stopSol()
```
[the remark this answers is not in the document]
Comment (Mahendra-intel) [reply src/app/devices/sol/sol.component.ts:119]: done

File: src/app/devices/sol/sol.component.ts [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+  cleanUp = (): void => {
+    this.terminal = null
+    this.redirector = null
+    this.dataProcessor = null
+    this.term = null
+  }
+
+  handleKeyPress = (domEvent: any): void => {
+    this.terminal.TermSendKeys(domEvent)
```
[the remark this answers is not in the document]
Comment (Mahendra-intel) [reply src/app/devices/sol/sol.component.ts:115]: in ui-toolkit => AMTTerminal.ts file it is declared as "TermSendKeys"

File: src/app/devices/sol/sol.component.ts (hunk 3 above)
Comment (Mahendra-intel) [reply src/app/devices/sol/sol.component.ts:14]: done

File: src/app/devices/sol/sol.component.ts (hunk 1 above)
Comment (Mahendra-intel) [reply src/app/devices/sol/sol.component.ts:22]: done

File: src/app/devices/sol/sol.component.ts [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+      fontWeight: 'bold'
+    })
+  }
+
+  handleWriteToXterm = (str: string): any => this.term.write(str)
+
+  handleClearTerminal = (): any => this.term.reset()
+
+  onSOLConnect = (e: Event): void => {
+    if (this.redirector !== 'null') {
```
Comment (rsdmike) [inline src/app/devices/sol/sol.component.ts:84]: this should be `if (this.redirector != null)`

**Review by rsdmike — COMMENTED** [review]
Looks good except for the condition, once thats updated i'll go ahead and merge

File: src/app/devices/sol/sol.component.ts (hunk 6 above)
Comment (Mahendra-intel) [reply src/app/devices/sol/sol.component.ts]: done

**Review by rsdmike — APPROVED** [review]

**Pull request merged**